frontend/packages/knative-plugin/src/utils/fetch-dynamic-eventsources-utils.ts

```diff
@@ -76,7 +77,14 @@ export const fetchEventSourcesCrd = async () => {
             namespaced: true,
             crd: true,
             color: knativeEventingColor.value,
-          });
+          };
+          const sourceIndex = _.findIndex(accumulator, ['kind', kind]);
+          // added check as some sources has multiple entries with deprecated APIgroups
+          if (sourceIndex === -1) {
```

Comment:
So if the non-deprecated version comes first... and then the deprecated version... you'd add it still? Are these sorted?

This plays into Rohit's comment as well...

Reply:
@andrewballantyne if non-deprecated version comes first it'll be added to `accumulator` and then if deprecated comes then it won't update the accumulator.

https://github.com/openshift/console/pull/5584/files#diff-10f33da55e6ecf19ae80b6c1c09b45a8R85

Added it upfront as there are cases where there just one APIGroup and it's deprecated updating it later if there are non deprecated ApiGroup

Reply:
Doh, misread the code... you're good haha.